fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1926,33 +1926,48 @@ public int reindex(FHIRPersistenceContext context, OperationOutcome.Builder oper
                     logicalId = parts[1];
                 }
 
-                // Look up the resourceTypeId for the given resourceType
+                // Look up the optional resourceTypeId for the given resourceType parameter
```

Comment:
what does it mean for the resourceTypeId to be optional?

Reply:
You can reindex:
1 all resource types
2 "Observation" - all Observation resources
3 "Observation/abc123" - a specific Observation resource

For 1, the resourceTypeId will be null.